File: src/main/java/net/sharplab/epubtranslator/core/driver/translator/DeepLClient.java
package net.sharplab.epubtranslator.core.driver.translator;

import javax.ws.rs.Consumes;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MultivaluedMap;
import java.io.Closeable;

@Path("/v2/translate")
public interface DeepLClient extends Closeable {

    @POST
    @Consumes("application/x-www-form-urlencoded")
    @Produces("application/json")
    DeepLTranslateAPIResponse translate(MultivaluedMap<String, String> request);
}

File: src/main/java/net/sharplab/epubtranslator/core/driver/translator/DeepLClient.kt
package net.sharplab.epubtranslator.core.driver.translator

import java.io.Closeable
import javax.ws.rs.Consumes
import javax.ws.rs.POST
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MultivaluedMap

@Path("/v2/translate")
interface DeepLClient : Closeable {
    @POST
    @Consumes("application/x-www-form-urlencoded")
    @Produces("application/json")
    fun translate(request: MultivaluedMap<String, String>): DeepLTranslateAPIResponse
}
